iceberg/iceberg-rest-server/src/main/java/org/apache/gravitino/iceberg/service/IcebergRequestContext.java

```diff
@@ -19,16 +19,23 @@
 
 package org.apache.gravitino.iceberg.service;
 
+import java.util.Map;
 import javax.servlet.http.HttpServletRequest;
 import lombok.Getter;
+import org.apache.gravitino.utils.PrincipalUtils;
 
 /** The general request context information for Iceberg REST operations. */
 public class IcebergRequestContext {
-  @Getter private final HttpServletRequest httpRequest;
+
   @Getter private final String catalogName;
+  @Getter private final String userName;
+  @Getter private final String remoteIp;
```

Comment:
remoteHostName and remoteIP contains same information, I perfer just keep one,  do you think remoteHostName is more useful than remoteIP?

Reply:
I think it should be made clear whether the return value is IP address or hostname, or others, it should not be mixed conceptually.

Besides, do we need a remote port as well?

Reply:
This class now becomes a public class that will be leveraged by users. You should move it to the `listener/api/event` package, also:

1. don't use lombok to generate the method, clearly define them manually and add javadoc.
2. make the class stable and make all the event compatible.

Reply:
updated with `remoteHostName`, because it's more user friendly, if it has some performance decrease for resolving hostname, let's change to `remoteIP`

Reply:
As to the remote port , seems not necessary,  databricks and HDFS audit doesn't contain the remote port too.

Reply:
@jerryshao  what do you mean `make the class stable and make all the event compatible.` ?

Reply:
keep old httpServeletRequest in context, so old listener could use the HTTP request in the event.